src/VecSim/spaces/IP/IP_AVX512_FP64.cpp

```diff
@@ -0,0 +1,149 @@
+#include "IP_AVX512.h"
+#include "VecSim/spaces/space_includes.h"
+#include "VecSim/spaces/IP/IP.h"
+
+double FP64_InnerProductSIMD8Ext_AVX512_impl(const void *pVect1v, const void *pVect2v, size_t qty) {
+    double *pVect1 = (double *)pVect1v;
+    double *pVect2 = (double *)pVect2v;
+
+    const double *pEnd1 = pVect1 + qty;
+
+    __m512d sum512 = _mm512_set1_pd(0);
+
+    // In each iteration we calculate 8 doubles = 512 bits.
+    while (pVect1 < pEnd1) {
+
+        __m512d v1 = _mm512_loadu_pd(pVect1);
+        pVect1 += 8;
+        __m512d v2 = _mm512_loadu_pd(pVect2);
+        pVect2 += 8;
+        sum512 = _mm512_add_pd(sum512, _mm512_mul_pd(v1, v2));
+    }
+
+    return _mm512_reduce_add_pd(sum512);
+}
+
+double FP64_InnerProductSIMD8Ext_AVX512(const void *pVect1, const void *pVect2, size_t qty) {
+    return 1.0 - FP64_InnerProductSIMD8Ext_AVX512_impl(pVect1, pVect2, qty);
+}
+
+double FP64_InnerProductSIMD2Ext_AVX512_impl(const void *pVect1v, const void *pVect2v, size_t qty) {
```

Comment:
I think that this one and its derivative should be checked with `#ifdef __AVX512DQ__`